File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/surveysetting/SurveyOrgMappingVO.java
package nl.ing.api.contacting.conf.domain.model.surveysetting;

import nl.ing.api.contacting.conf.domain.validation.SurverySettingValidation;
import nl.ing.api.contacting.java.domain.OrganisationVO;

import java.util.Optional;

public record SurveyOrgMappingVO(
        Optional<Long> id,
        long surveyId,
        OrganisationVO organisationVO
) {
    public SurveyOrgMappingVO {
        SurverySettingValidation.validateId(id);
        SurverySettingValidation.validateSurveyId(surveyId);
        SurverySettingValidation.validateOrganisationVO(organisationVO);
    }
}

File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/domain/SurveyOrgMappingVO.scala
package nl.ing.api.contacting.conf.domain

import nl.ing.api.contacting.conf.domain.types.ContactingTypes.DatabaseId
import nl.ing.api.contacting.domain.slick.OrganisationVO

case class SurveyOrgMappingVO(id: Option[DatabaseId], surveyId: DatabaseId, organisationVO: OrganisationVO)